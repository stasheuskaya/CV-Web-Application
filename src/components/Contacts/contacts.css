.contactsContainer button {
    display: flex;
    background: white;
    border: none;
    color: var(--green-color-highlight-shamrock);
    font-size: 28px;
    
  }
  a:link {
    text-decoration: none;
    color: var(--main-text-and-bg-color-oxford-blue);
  }  
  
  a:visited {
    color: var(--main-text-and-bg-color-oxford-blue);
  }
  a:active {
    text-decoration: underline;
  }
  
  .contactBox {
    text-align: left;
  }
  
  .contact th {
    padding-right: 20px;
  }
  
  .contact h1, .contactBoundle h2 {
    padding-bottom: 15px;
  }
  
  .contactImg {
    vertical-align: -webkit-baseline-middle;
  }