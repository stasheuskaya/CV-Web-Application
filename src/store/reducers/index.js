import { combineReducers } from "redux";
import { skillsReducer } from "./skills_reducer";
import { educationsReducer } from "./educations_reducer";

const allReducers = combineReducers({
    skills: skillsReducer,
    educations: educationsReducer
})

export default allReducers;