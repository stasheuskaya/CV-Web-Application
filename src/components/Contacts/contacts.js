import './contacts.css';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faEnvelope } from '@fortawesome/free-solid-svg-icons';
import { faPhone } from '@fortawesome/free-solid-svg-icons';
import twitterIcon from '../../assets/images/twitterIcon.png';
import facebookIcon from '../../assets/images/facebookIcon.png';
import skypeIcon from '../../assets/images/skypeIcon.png';
import Button from '../Button/button';



export default function Contacts (props) {
    return (
        <div className="contactsContainer"> 
             <table className="contactBox">
                <tbody className="contact">
                    <th>
                        <Button icon={<FontAwesomeIcon icon={faPhone} />}/>
                    </th>
                    <th>
                        <h1><a href='tel:500-342-0242'>500 342 242</a></h1>
                    </th>
                </tbody>
                <tbody  className="contact">
                    <th>
                        <Button icon={<FontAwesomeIcon icon={faEnvelope} onClick={() => window.location = 'mailto:yourmail@gmail.com'} />}/> 
                    </th>
                    <th>
                        <h1><a href="mailto:yourmail@gmail.com">y.stashevskaya@gmail.com</a></h1>
                    </th>
                </tbody>
                <tbody className="contactBoundle">
                    <th>
                        <Button icon={<img src={twitterIcon} className="contactImg" alt="twitter"/>} /> 
                    </th>
                    <th>
                        <div>
                            <h1>Twitter</h1> <h2><a href="https://www.twitter.com">https://www.twitter.com</a></h2>
                        </div>
                    </th>
                </tbody>
                <tbody className="contactBoundle">
                    <th>
                        <Button icon={<img src={facebookIcon} className="contactImg" alt="facebook" />}/>
                    </th>
                    <th>
                        <div>
                            <h1>facebook</h1>
                            <h2><a href="https://www.facebook.com/facebook">https://www.facebook.com/facebook</a></h2>                
                        </div>
                    </th>
                </tbody>
                <tbody>
                    <th>
                        <Button icon={<img src={skypeIcon} className="contactImg" alt="skype"/>}/>
                    </th>
                    <th>
                        <div>
                            <h1>Skype</h1>
                            <h2><a href="skype:<username>?<action>">https://www.skype.com</a></h2>  
                        </div>
                    </th>
                </tbody>
            </table>
        </div>
    )
}
