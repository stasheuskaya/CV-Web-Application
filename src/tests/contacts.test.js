import renderer from "react-test-renderer";
import Contacts from "../components/Contacts/contacts";
import { Provider } from 'react-redux';
import { store } from "../store/store"

describe('Contacts Page', () => {
    it('renders content', () => {
		expect(renderer.create(
			<Provider store={store}>
                <Contacts />
			</Provider>)).toMatchSnapshot();
	});
});