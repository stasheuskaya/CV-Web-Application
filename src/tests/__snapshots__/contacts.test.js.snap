// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contacts Page renders content 1`] = `
<div
  className="contactsContainer"
>
  <table
    className="contactBox"
  >
    <tbody
      className="contact"
    >
      <th>
        <button
          icon={
            <FontAwesomeIcon
              beat={false}
              beatFade={false}
              border={false}
              bounce={false}
              className=""
              fade={false}
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    512,
                    512,
                    Array [
                      128379,
                      128222,
                    ],
                    "f095",
                    "M511.2 387l-23.25 100.8c-3.266 14.25-15.79 24.22-30.46 24.22C205.2 512 0 306.8 0 54.5c0-14.66 9.969-27.2 24.22-30.45l100.8-23.25C139.7-2.602 154.7 5.018 160.8 18.92l46.52 108.5c5.438 12.78 1.77 27.67-8.98 36.45L144.5 207.1c33.98 69.22 90.26 125.5 159.5 159.5l44.08-53.8c8.688-10.78 23.69-14.51 36.47-8.975l108.5 46.51C506.1 357.2 514.6 372.4 511.2 387z",
                  ],
                  "iconName": "phone",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              maskId={null}
              pull={null}
              pulse={false}
              rotation={null}
              shake={false}
              size={null}
              spin={false}
              swapOpacity={false}
              symbol={false}
              title=""
              titleId={null}
              transform={null}
            />
          }
        >
          <div
            className="icon"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-phone "
              data-icon="phone"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M511.2 387l-23.25 100.8c-3.266 14.25-15.79 24.22-30.46 24.22C205.2 512 0 306.8 0 54.5c0-14.66 9.969-27.2 24.22-30.45l100.8-23.25C139.7-2.602 154.7 5.018 160.8 18.92l46.52 108.5c5.438 12.78 1.77 27.67-8.98 36.45L144.5 207.1c33.98 69.22 90.26 125.5 159.5 159.5l44.08-53.8c8.688-10.78 23.69-14.51 36.47-8.975l108.5 46.51C506.1 357.2 514.6 372.4 511.2 387z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </div>
        </button>
      </th>
      <th>
        <h1>
          <a
            href="tel:500-342-0242"
          >
            500 342 242
          </a>
        </h1>
      </th>
    </tbody>
    <tbody
      className="contact"
    >
      <th>
        <button
          icon={
            <FontAwesomeIcon
              beat={false}
              beatFade={false}
              border={false}
              bounce={false}
              className=""
              fade={false}
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    512,
                    512,
                    Array [
                      128386,
                      61443,
                      9993,
                    ],
                    "f0e0",
                    "M464 64C490.5 64 512 85.49 512 112C512 127.1 504.9 141.3 492.8 150.4L275.2 313.6C263.8 322.1 248.2 322.1 236.8 313.6L19.2 150.4C7.113 141.3 0 127.1 0 112C0 85.49 21.49 64 48 64H464zM217.6 339.2C240.4 356.3 271.6 356.3 294.4 339.2L512 176V384C512 419.3 483.3 448 448 448H64C28.65 448 0 419.3 0 384V176L217.6 339.2z",
                  ],
                  "iconName": "envelope",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              maskId={null}
              onClick={[Function]}
              pull={null}
              pulse={false}
              rotation={null}
              shake={false}
              size={null}
              spin={false}
              swapOpacity={false}
              symbol={false}
              title=""
              titleId={null}
              transform={null}
            />
          }
        >
          <div
            className="icon"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-envelope "
              data-icon="envelope"
              data-prefix="fas"
              focusable="false"
              onClick={[Function]}
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M464 64C490.5 64 512 85.49 512 112C512 127.1 504.9 141.3 492.8 150.4L275.2 313.6C263.8 322.1 248.2 322.1 236.8 313.6L19.2 150.4C7.113 141.3 0 127.1 0 112C0 85.49 21.49 64 48 64H464zM217.6 339.2C240.4 356.3 271.6 356.3 294.4 339.2L512 176V384C512 419.3 483.3 448 448 448H64C28.65 448 0 419.3 0 384V176L217.6 339.2z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </div>
        </button>
      </th>
      <th>
        <h1>
          <a
            href="mailto:yourmail@gmail.com"
          >
            y.stashevskaya@gmail.com
          </a>
        </h1>
      </th>
    </tbody>
    <tbody
      className="contactBoundle"
    >
      <th>
        <button
          icon={
            <img
              alt="twitter"
              className="contactImg"
              src="twitterIcon.png"
            />
          }
        >
          <div
            className="icon"
          >
            <img
              alt="twitter"
              className="contactImg"
              src="twitterIcon.png"
            />
          </div>
        </button>
      </th>
      <th>
        <div>
          <h1>
            Twitter
          </h1>
           
          <h2>
            <a
              href="https://www.twitter.com"
            >
              https://www.twitter.com
            </a>
          </h2>
        </div>
      </th>
    </tbody>
    <tbody
      className="contactBoundle"
    >
      <th>
        <button
          icon={
            <img
              alt="facebook"
              className="contactImg"
              src="facebookIcon.png"
            />
          }
        >
          <div
            className="icon"
          >
            <img
              alt="facebook"
              className="contactImg"
              src="facebookIcon.png"
            />
          </div>
        </button>
      </th>
      <th>
        <div>
          <h1>
            facebook
          </h1>
          <h2>
            <a
              href="https://www.facebook.com/facebook"
            >
              https://www.facebook.com/facebook
            </a>
          </h2>
        </div>
      </th>
    </tbody>
    <tbody>
      <th>
        <button
          icon={
            <img
              alt="skype"
              className="contactImg"
              src="skypeIcon.png"
            />
          }
        >
          <div
            className="icon"
          >
            <img
              alt="skype"
              className="contactImg"
              src="skypeIcon.png"
            />
          </div>
        </button>
      </th>
      <th>
        <div>
          <h1>
            Skype
          </h1>
          <h2>
            <a
              href="skype:<username>?<action>"
            >
              https://www.skype.com
            </a>
          </h2>
        </div>
      </th>
    </tbody>
  </table>
</div>
`;
